app/src/main/java/com/example/quizacademytask/StackListFragment.kt

```diff
@@ -122,49 +126,28 @@ class StackListFragment : Fragment(), SimpleAdapter.OnItemClickListener {
         stacksAdapter = SimpleAdapter(stacksList, this)
     }
 
-
     /* Create a Course from JSON */
     private fun createCourse(jsonString: String): Course {
         var course: Course? = null
         runBlocking {
             launch {
-                courseObj = Gson().fromJson(jsonString, CourseObject::class.java)
```

Comment:
you could extract parsing to function or helper class to hide these implementation details